### DISCUSSION TITLE
Neo-tree git status not refreshing automatially

### DISCUSSION THREAD

**Discussion by benjaminbauer** [LazyVim/LazyVim#5079] [Q&A]
When I change/add files in a git repo, neo-tree correctly reflects the changes in the file tree. However when I stage/commit the changes, I need to manually refresh or toggle neo-tree.

This discussion seems to imply, that it is a problem in conjunction with edgy: https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/1253

Is there something wrong with my setup or is this the default behaviour in LayzVim? In the latter case: I think it would be great, if this would work out of the box.

**Comment by dpetka2001** [comment]
It seems to work correctly for me. Screencast following

[Screencast_2024-12-14-22-17-07.webm](https://github.com/user-attachments/assets/1cce75fb-2c0b-4f4f-b674-917000c862b3)

**Comment by bcardarella** [comment]
I am also having this issue. I'm on MacOS using Ghostty.

I added the following to init.lua:

```lua
require("neo-tree").setup({
  enable_git_status = true, -- Ensure Git status is enabled
  filesystem = {
    follow_current_file = {
        enabled = true
    }, -- Optional, follows the currently active buffer
    use_libuv_file_watcher = true, -- Enables the file watcher
  },
})
```

and this does work when I stage a file. But when I unstage it Neotree doesn't update.

**Reply by dpetka2001** [reply]
How do you stage/unstage? If i use lazygit inside Neovim, then neo-tree gets updated correctly. If you use `git add` from terminal, then this indeed doesn't work as also mentioned in this [Neotree issue](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/724). You have to also put in the setup `git_status_async = false` in that case and make sure that the `git_root == neotree_root` as also mentioned in [this comment](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/724#issuecomment-1418023126).

**Reply by bcardarella** [reply]
terminal

**Reply by bcardarella** [reply]
in this case the git_root is neotree_root

**Reply by bcardarella** [reply]
actually, changing `git_status_async = false` did fix it, thanks!
